graphql/resolvers/exerciseCrud.ts

```diff
@@ -91,3 +92,67 @@ export const deleteExercise = async (
     }
   })
 }
+
+export const flagExercise = async (
+  _parent: void,
+  arg: MutationFlagExerciseArgs,
+  { req }: Context
+): Promise<Exercise> => {
+  const { id } = arg
+
+  const authorId = req.user?.id
+  if (!authorId) throw new Error('No User')
+
+  const exercise = await prisma.exercise.findUnique({
+    where: {
+      id
+    }
+  })
+
+  if (!isAdmin(req) && exercise?.authorId !== authorId) {
+    throw new Error('Not authorized to flag')
+  }
+
+  return prisma.exercise.update({
+    where: { id },
+    data: {
+      flaggedAt: new Date().toISOString()
+    },
+    include: {
+      author: true,
+      module: true
+    }
+  })
+}
+
+export const removeExerciseFlag = async (
+  _parent: void,
+  arg: MutationRemoveExerciseFlagArgs,
+  { req }: Context
+): Promise<Exercise> => {
+  const { id } = arg
+
+  const authorId = req.user?.id
+  if (!authorId) throw new Error('No User')
+
+  const exercise = await prisma.exercise.findUnique({
+    where: {
+      id
+    }
+  })
+
+  if (!isAdmin(req) && exercise?.authorId !== authorId) {
```

Comment:
it does not make sense to only allow the creator of the exercise to unflag his exercise.

Admins can unflag any exercise